### ISSUE TITLE
DynamoDB Enhanced Client : Nested documents in collections not working correctly in BeanTableSchema

### ISSUE THREAD

**Issue by bmaizels** [aws/aws-sdk-java-v2#1748]
## Describe the bug
BeanTableSchema does not correctly convert document beans when they are used in a collection (eg: List<DocumentBean>, Map<String, DocumentBean>). 

## Expected Behavior
These items should be converted correctly.

## Current Behavior
An exception is thrown stating a converter cannot be found for the DocumentBean class.

## Repro
```
    @Test
    public void documentBean_list_correctlyMapsAttributes() {
        BeanTableSchema<DocumentBean> beanTableSchema = BeanTableSchema.create(DocumentBean.class);
        AbstractBean abstractBean1 = new AbstractBean();
        abstractBean1.setAttribute2("two");
        AbstractBean abstractBean2 = new AbstractBean();
        abstractBean2.setAttribute2("three");
        DocumentBean documentBean = new DocumentBean();
        documentBean.setId("id-value");
        documentBean.setAttribute1("one");
        documentBean.setAbstractBeanList(Arrays.asList(abstractBean1, abstractBean2));

        AttributeValue expectedDocument1 = AttributeValue.builder()
                                                        .m(singletonMap("attribute2", stringValue("two")))
                                                        .build();
        AttributeValue expectedDocument2 = AttributeValue.builder()
                                                         .m(singletonMap("attribute2", stringValue("three")))
                                                         .build();
        AttributeValue expectedList = AttributeValue.builder().l(expectedDocument1, expectedDocument2).build();

        Map<String, AttributeValue> itemMap = beanTableSchema.itemToMap(documentBean, false);
        assertThat(itemMap.size(), is(3));
        assertThat(itemMap, hasEntry("id", stringValue("id-value")));
        assertThat(itemMap, hasEntry("attribute1", stringValue("one")));
        assertThat(itemMap, hasEntry("abstractBeanList", expectedList));
    }
```

**Comment by juanqui** [comment]
Thanks for the quick fix on this! I pulled your branch, built new dynamodb and dynamodb-enhanced packages, updated our project to use them, and I was able to ran both our unit and integ tests successfully. Previously, our unit test was failing due to the exception that was being thrown by the `BeanTableSchema` (as report [here](https://github.com/aws/aws-sdk-java-v2/issues/35). Our integ tests runs against a live database, and passed without a problem.

Took a look at your implementation and it looks great. Was specifically looking for the recursion :)

Thanks again!
